dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops

```diff
@@ -732,3 +732,26 @@ CWWKD1076.repo.disposed.explanation=Repository methods must not be used outside
  of the lifecycle of the application that defines the repository.
 CWWKD1076.repo.disposed.useraction=Ensure that the repository method is not \
  used outside of the scope of the application that defines the repository.
+
+CWWKD1077.defaultds.not.found=CWWKD1077E: The {0} application artifact \
+ has a {1} repository interface that uses or defaults to the \
+ {2} data store, which must be configured in server configuration \
+ by defining a dataSource element that has an identifier of \
+ DefaultDataSource. For example, {3} \
+ Alternatively, assign the dataStore value of the Repository annotation \
+ to be the identifier of a databaseStore element or the JNDI name of \
+ a data source resource reference or persistence unit reference \
+ that is accessible to the application artifact. A databaseStore element \
+ can be defined in server configuration. For example, {4} \
+ An application component can use the Resource annotation to define a \
+ data source resource reference. For example, {5}, which requires server \
+ configuration to have a dataSource configuration element with a {6} attribute. \
+ An application component can use the PersistenceUnit annotation to define a \
+ persistence unit reference. For example, {7}, which requires the persistence.xml \
+ file to have a persistence-unit element with a matching name attribute. \
+ For example, {8}
```

Comment:
Note to ID reviewers - The new message intentionally omits the . after some of the examples where it would appear awkward along with the substituted values. This is in response to feedback from my team when reviewing the message output. Hopefully this is acceptable.

The message itself is lengthy because there is a lot of information to convey. It is a very likely error for new users, who will need the information in this message to get things working.

Reply:
I see data source mentioned like -> dataSource and data source -> Which would be the correct usage?

Reply:
> I see data source mentioned like -> dataSource and data source -> Which would be the correct usage?

They are two different things. `<dataSource>` is the name of a configuration element in server.xml.  When referring to this in the message, I used "dataSource element" or "dataSource configuration element". 
The term "data source" refers to the concept of a data source that the application interacts with, either in terms of a data source resource reference or a `javax.sql.DataSource` instance in their code.  Let me know if you would like me to change "data source" to `DataSource` to be more explicit.  The term `dataSource` must remain as is because it directly names the configuration element.